### DISCUSSION TITLE
Is there an animation that draws the Linechart from left to right (Horizontally)?

### DISCUSSION THREAD

**Discussion by abderhim** [patrykandpatrick/vico#351] [Ideas]

**Comment by Gowsky** [comment]
Hello @abderhim,
There’s no animation like that currently, but we will consider adding it.

**Reply by RandunuK** [reply]
Hi, is there any update on this?  Thanks for the great library.

**Reply by Gowsky** [reply]
Hello, there is no update on this. We will share updates here when we have them.

**Comment by nagatsuka-shuuya** [comment]
@Gowsky I've raised the [PR](https://github.com/patrykandpatrick/vico/pull/1483) that implements this feature. Could you check please?

**Comment by claude** [comment]
This is now available in [3.3.0](https://github.com/patrykandpatrick/vico/releases/tag/v3.3.0).

`LineCartesianLayer` has a reveal animation. Pass `CartesianLayerDrawingModelInterpolator.line(sweep = true)` as the layer’s `drawingModelInterpolator`, and the line’s first appearance is animated as a clip sweeping in from the start edge at full opacity and _y_, rather than the default grow-from-baseline fade-in:

```kotlin
rememberCartesianChart(
  rememberLineCartesianLayer(
    drawingModelInterpolator = CartesianLayerDrawingModelInterpolator.line(sweep = true)
  ),
  startAxis = VerticalAxis.rememberStart(),
  bottomAxis = HorizontalAxis.rememberBottom(),
)
```

The duration and easing come from `CartesianChartHost`’s new `initialAnimationSpec` parameter, which defaults to `animationSpec`. See `RevealLineChart` in the sample sources for a complete example.

Thanks for the suggestion, and sorry it took a while.

<sub>Based on maintainer input.</sub>
